ProcessesApi/V1/Services/SoleToJointService.cs

```diff
@@ -172,6 +173,29 @@ private async Task OnProcessCancelled(Stateless.StateMachine<string, string>.Tra
             await PublishProcessClosedEvent(x).ConfigureAwait(false);
         }
 
+        private async Task OnProcessCompleted(Stateless.StateMachine<string, string>.Transition x)
+        {
+            var processRequest = x.Parameters[0] as ProcessTrigger;
```

Comment:
Is this similar to OnProcessClosed? Maybe you could move most of the logic into a helper method?